gluten-substrait/src/main/scala/org/apache/gluten/execution/ScanTransformerFactory.scala

```diff
@@ -25,11 +25,17 @@ object ScanTransformerFactory {
 
   def createFileSourceScanTransformer(
       scanExec: FileSourceScanExec): FileSourceScanExecTransformerBase = {
+    // Partition filters will be evaluated in driver side, so we can remove
+    // GlutenTaskOnlyExpressions
+    val partitionFilters = scanExec.partitionFilters.filter {
+      case _: GlutenTaskOnlyExpression => false
+      case _ => true
+    }
     FileSourceScanExecTransformer(
       scanExec.relation,
       scanExec.output,
       scanExec.requiredSchema,
-      scanExec.partitionFilters,
+      partitionFilters,
```

Comment:
Do you need to consider this issue also for the BatchScanTransformer? Thanks.

Reply:
> Do you need to consider this issue also for the BatchScanTransformer? Thanks.

There are no partition filters in `BatchScanExecTransformer`, and runtime filters will also be converted to source v2 predicates, so there is no similar issue for ``BatchScanExecTransformer`.